provisioner.go

```diff
@@ -177,13 +179,30 @@ func (p *LocalPathProvisioner) getRandomPathOnNode(node string) (string, error)
 	return path, nil
 }
 
+func (p *LocalPathProvisioner) isMultiNodeAlowable() (bool, error) {
+	p.configMutex.RLock()
+	defer p.configMutex.RUnlock()
+
+	if p.config == nil {
+		return false, fmt.Errorf("no valid config available")
+	}
+
+	c := p.config
+
+	return c.AllowMultiNode, nil
+}
+
 func (p *LocalPathProvisioner) Provision(opts pvController.ProvisionOptions) (*v1.PersistentVolume, error) {
 	pvc := opts.PVC
 	if pvc.Spec.Selector != nil {
 		return nil, fmt.Errorf("claim.Spec.Selector is not supported")
 	}
+	allowMultiNode, err := p.isMultiNodeAlowable()
+	if err != nil {
+		return nil, err
+	}
 	for _, accessMode := range pvc.Spec.AccessModes {
-		if accessMode != v1.ReadWriteOnce {
+		if (accessMode != v1.ReadWriteOnce) && (!allowMultiNode) {
```

Comment:
Within the current PR both scripts are working just fine. Because from their point of view shared filesystem are local (it is mounted). But I realized the current PR is not getting the best from the network filesytem.

The path can be created on every node of the Kubernetes cluster and can be deleted on every node as well. Pod can be evacuated to other node and still have access to the data.
So I think it can be done the following way:
- `sig-storage-lib-external-provisioner` will pick up a random node.
-  if `allowMultiNode: true` this node can be written not in `spec/nodeAffinity` (where getPathAndNodeForPV will find it later), but in metadata instead.

Or information about initial node can be lost completely after creation, so on deletion another random node can be chosen. The latter is even more safe, because node which created the path can be broken/drained/absent on the time of deletion.